engine/access/rpc/backend/backend_scripts.go

```diff
@@ -81,6 +82,31 @@ func (b *backendScripts) ExecuteScriptAtBlockHeight(
 	return b.executeScriptOnExecutionNode(ctx, blockID, script, arguments)
 }
 
+func findScriptExecutors(
```

Comment:
This function is the main change, as well as the function that calls it.

It wraps the original `executionNodesForBlockID`, and check whether archiveAddress is configured, which will take the priority to be used as script executor.